hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -1868,12 +1868,12 @@ public boolean isDefinedOnSubclass(int i) {
 	}
 
 	@Override
-	public int[] resolveAttributeIndexes(Set<String> properties) {
-		Iterator<String> iter = properties.iterator();
-		int[] fields = new int[properties.size()];
+	public int[] resolveAttributeIndexes(String[] attributeNames) {
+		int[] fields = new int[attributeNames.length];
```

Comment:
Ultimately the fields returned here need to be unique.  This used to be enforced by passing in the Set, but since that is no longer the case you'll have to account for that in your code here.